router/router.go

```diff
@@ -161,155 +159,108 @@ func (rt *HandleT) workerProcess(worker *workerT) {
 
 		var isPrevFailedUser bool
 		var previousFailedJobID int64
-		if canEventBeMappedToUser {
+		if canEventBeMappedToUser && rt.keepOrderOnFailure {
 			//If there is a failed jobID from this user, we cannot pass future jobs
 			worker.failedJobIDMutex.RLock()
 			previousFailedJobID, isPrevFailedUser = worker.failedJobIDMap[userID]
 			worker.failedJobIDMutex.RUnlock()
 
-			if isPrevFailedUser && previousFailedJobID < job.JobID {
-				logger.Debugf("[%v Router] :: skipping processing job for userID: %v since prev failed job exists, prev id %v, current id %v", rt.destID, userID, previousFailedJobID, job.JobID)
-				resp := fmt.Sprintf(`{"blocking_id":"%v", "user_id":"%s"}`, previousFailedJobID, userID)
-				status := jobsdb.JobStatusT{
-					JobID:         job.JobID,
-					AttemptNum:    job.LastJobStatus.AttemptNum,
-					ExecTime:      time.Now(),
-					RetryTime:     time.Now(),
-					ErrorCode:     respStatus,
-					JobState:      jobsdb.Waiting.State,
-					ErrorResponse: []byte(resp), // check
-				}
-				rt.responseQ <- jobResponseT{status: &status, worker: worker, userID: userID}
+			// mark job as waiting if prev job from same user has not succeeded yet
+			skipSending := isPrevFailedUser && rt.handlePrevFailedUserJob(job, paramaters, userID, worker, previousFailedJobID)
```

Comment:
This is a bit difficult to read. Let's use `if` instead of `&&` .


```
func (rt *HandleT) handlePrevFailedUserJob(job *jobsdb.JobT, paramaters ParametersT, userID string, worker *workerT, previousFailedJobID int64) (markedWaiting bool, error) {
    
}
```

```
if isPrevFailedUser {
   markedWaiting, err := rt.handlePrevFailedUserJob(job, paramaters, userID, worker, previousFailedJobID)
   if err != nil {
       panic(err)
   }
   if markedWaiiting {
      continue
   }
}
```